pkg/audit/manager.go

```diff
@@ -375,6 +376,18 @@ func (am *Manager) getUpdateListsFromAuditResponses(res []*constraintTypes.Resul
 		ea := util.EnforcementAction(enforcementAction)
 		totalViolationsPerEnforcementAction[ea]++
 		logViolation(am.log, r.Constraint, r.EnforcementAction, result)
+		if *detailedViolationsMetrics {
+			err := am.reporter.reportViolationsDetails(
+				r.EnforcementAction,
+				r.Constraint.GetKind(),
+				r.Constraint.GetName(),
+				result.rkind,
```

Comment:
Having the object name persisted in metrics gives any user the ability to OOM the server by spamming it with a bunch of invalid resources. This could break audit.

That being said, any scaling of audit results beyond the memory capacity of Gatekeeper could have this effect, but the above attack is unique in that by creating/deleting objects, a user could break audit without impacting the API server.

The result of such an attack should be the audit server restarting with audit metrics wiped. I'm not sure how well-behaved the server would be once it comes under memory stress. It's also an open question how well-behaved any downstream prometheus aggregators would be.